docs/data-cleanup-mechanisms.md

```diff
@@ -0,0 +1,65 @@
+# Data cleanup mechanisms
+
+There are several mechanisms regarding the cleanup of data in EmailAlertApi.
+This doc gives a brief overview of each.
+
+## Nullifying subscribers
```

Comment:
I wonder if it's worth changing the word subscribers to email addresses. The language of nullifying a subscriber feels a bit odd and it's the email address we're doing this to not the subscriber.

Reply:
Hmm, that's a good point!

I always thought of it through database goggles as a subscriber record has an email address and so we're effectively nullifying the subscriber. However I can see that doesn't make much sense to non devs so I've updated it to `Nullifying email addresses`.